ui/core/components/pickers/list_picker.tsx

```diff
@@ -274,30 +318,73 @@ export class ListPicker<ModObject, ItemType> extends Input<ModObject, Array<Item
 			itemContainer.addEventListener(
 				'dragover',
 				event => {
-					if (!curDragData || curDragData.listPicker != this) {
+					if (!curDragData || curDragData.listPicker.config.itemLabel !== this.config.itemLabel) {
+						return;
+					}
+					if (this.config.itemLabel === 'Action' && curDragData.listPicker !== this) {
+						event.dataTransfer!.dropEffect = 'none';
 						return;
 					}
+					event.dataTransfer!.dropEffect = 'move';
+					event.stopPropagation();
 					event.preventDefault();
 				},
 				{ signal: this.signal },
 			);
 
+			itemContainer.addEventListener(
+				'dragend',
+				event => {
+					if (
+						!curDragData ||
+						curDragData.listPicker.config.itemLabel !== this.config.itemLabel ||
+						(this.config.itemLabel === 'Action' && curDragData.listPicker !== this)
+					) {
+						return;
+					}
+					event.stopPropagation();
+					itemContainer.removeAttribute('draggable');
+					curDragData.item.elem.removeAttribute('draggable');
+					curDragData.item.elem.classList.remove('dragfrom');
+					[...document.querySelectorAll('.dragto,.hover')].forEach(elem => {
```

Comment:
Getting these specific values can probably also become a getter/util

Reply:
Do you mean the classes? Like a helper to fetch any number of elements matching certain classes and then remove the classes from the matching elements?
A very niche use case :D

Reply:
Cleaned up some repetitive code, this one being one of them